native/src/main/java/io/ballerina/lib/data/xmldata/utils/DataUtils.java

```diff
@@ -782,6 +765,27 @@ private static String addAttributeToRecord(BString prefix, BString uri, String k
         return prefix.getValue().concat(Constants.COLON).concat(key);
     }
 
+    public static boolean isRegExpType(Type type) {
+        Module module = type.getPackage();
+        if (module == null) {
+            return false;
+        }
+
+        String moduleName = module.getName();
+        String typeName = type.getName();
+        if (typeName == null || moduleName == null) {
+            return false;
+        }
+        if (moduleName.equals(Constants.REGEXP_MODULE_NAME) && typeName.equals(Constants.REGEXP_TYPE_NAME)) {
+            return true;
+        }
+
+        if (type.getTag() == TypeTags.TYPE_REFERENCED_TYPE_TAG) {
```

Comment:
instead of doing recursion, shall we get the refered type in L769

Reply:
The issue here is due to the `regexp-type`. The type of regexp is considered as reference type of anydata and reladonly instead of subtype of it. Due to that, we can not detect that specific case if we get the referred type.